zerver/views/users.py

```diff
@@ -380,7 +380,7 @@ def add_bot_backend(
     if bot_type in (UserProfile.INCOMING_WEBHOOK_BOT, UserProfile.EMBEDDED_BOT) and service_name:
         check_valid_bot_config(bot_type, service_name, config_data)
 
-    bot_profile = do_create_user(email=email, password=None,
+    bot_profile = do_create_user(email=email, password='',
```

Comment:
This change should probably be extracted to a separate commit -- I don't understand why we'd want to change the password from `None`, which as I understand it means "no password" to `''`.